test/dlc_tests/sanity/test_security_check.py

```diff
@@ -35,6 +37,21 @@ def test_security(image):
         run(f"docker rm -f {container_name}", hide=True)
 
 
+def run_scan(ecr_client, image):
+    scan_status = None
+    start_time = time()
+    ecr_utils.start_ecr_image_scan(ecr_client, image)
+    while (time() - start_time) <= 600:
+        scan_status, scan_status_description = ecr_utils.get_ecr_image_scan_status(ecr_client, image)
+        if scan_status == "FAILED" or scan_status not in [None, "IN_PROGRESS", "COMPLETE"]:
+            raise ECRScanFailureException(f"ECR Scan failed for {image} with description: {scan_status_description}")
+        if scan_status == "COMPLETE":
+            break
+        sleep(1)
```

Comment:
it would be lot of API calls if we wait for 1 sec. Before making first call to check get_ecr_image_scan_status, we can wait for 3-5 mins based on how long ECR scan takes on average, then reduce wait time to check for every 1 min.

Or we can check for every 1 min always.

Reply:
I haven't seen any throttling on ECR API calls caused by these requests sent every second, which is why I've left it as is.

This implementation is sub-optimal, and can use Boto3 Waiters, but that is something we had planned to implement at a later date. I would like to limit the changes on this PR, for now, to not blow its scope up.